crates/build/src/lib.rs

```diff
@@ -764,18 +778,79 @@ fn post_process_wasm(
     optimized_code: &PathBuf,
     skip_wasm_validation: bool,
     verbosity: &Verbosity,
-    max_memory_pages: u32,
+    max_memory_pages: u64,
 ) -> Result<()> {
     // Deserialize Wasm module from a file.
-    let mut module =
+    let module =
         load_module(optimized_code).context("Loading of optimized wasm failed")?;
+    let mut output = Vec::new();
+    post_process_module(
```

Comment:
Maybe `post_process_module` could return an allocated `Vec<u8>` since it looks like every usage just passes a new buffer instance.